pystac/extensions/raster.py

```diff
@@ -0,0 +1,737 @@
+            Optional[float]
+        """
+        return self.properties.get("minimum")
+
+    @minimum.setter
+    def minimum(self, v: Optional[float]) -> None:
+        if v is not None:
+            self.properties["minimum"] = v
+        else:
+            self.properties.pop("minimum", None)
+
+    @property
+    def maximum(self) -> Optional[float]:
+        """Get or sets the maximum pixel value
+
+        Returns:
+            Optional[float]
+        """
+        return self.properties.get("maximum")
+
+    @maximum.setter
+    def maximum(self, v: Optional[float]) -> None:
+        if v is not None:
+            self.properties["maximum"] = v
+        else:
+            self.properties.pop("maximum", None)
+
+    @property
+    def mean(self) -> Optional[float]:
+        """Get or sets the mean pixel value
+
+        Returns:
+            Optional[float]
+        """
+        return self.properties.get("mean")
+
+    @mean.setter
+    def mean(self, v: Optional[float]) -> None:
+        if v is not None:
+            self.properties["mean"] = v
+        else:
+            self.properties.pop("mean", None)
+
+    @property
+    def stddev(self) -> Optional[float]:
+        """Get or sets the standard deviation pixel value
+
+        Returns:
+            Optional[float]
+        """
+        return self.properties.get("stddev")
+
+    @stddev.setter
+    def stddev(self, v: Optional[float]) -> None:
+        if v is not None:
+            self.properties["stddev"] = v
+        else:
+            self.properties.pop("stddev", None)
+
+    @property
+    def valid_percent(self) -> Optional[float]:
+        """Get or sets the Percentage of valid (not nodata) pixel
+
+        Returns:
+            Optional[float]
+        """
+        return self.properties.get("valid_percent")
+
+    @valid_percent.setter
+    def valid_percent(self, v: Optional[float]) -> None:
+        if v is not None:
+            self.properties["valid_percent"] = v
+        else:
+            self.properties.pop("valid_percent", None)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Returns the dictionary representing the JSON of those Statistics.
+
+        Returns:
+            dict: The wrapped dict of the Statistics that can be written out as JSON.
+        """
+        return self.properties
+
+    @staticmethod
+    def from_dict(d: Dict[str, Any]) -> "Statistics":
+        """Constructs an Statistics from a dict.
+
+        Returns:
+            Statistics: The Statistics deserialized from the JSON dict.
+        """
+        return Statistics(properties=d)
+
+
+class Histogram:
+    """Represents pixel distribution information attached to a band in the raster extension.
+
+    Use Band.create to create a new Band.
+    """
+
+    def __init__(self, properties: Dict[str, Any]) -> None:
+        self.properties = properties
+
+    def apply(
+        self,
+        count: int,
+        min: float,
+        max: float,
+        buckets: List[int],
```

Comment:
My interpretation of the [Histogram Object](https://github.com/stac-extensions/raster#histogram-object) in the extension docs is that all of these fields are optional since none are marked as **REQUIRED**. If that's correct, we should probably change these to optional types (e.g. `count: Optional[int]`), update the types of the associated properties below, and update the types of the `create` method arguments as well.

Reply:
Sorry, meant to just leave this as a comment instead of requesting changes. If they are in fact required we can leave it as-is.

Reply:
An histogram object is atomic and all fields are required for its interpretation. That's why I removed all the optional types. I will update the extension documentation to clarify this.

Reply:
https://github.com/stac-extensions/raster/pull/15

Reply:
👍